### ISSUE TITLE
`jsons.dumps(np.array([0]))` reaches maximum recursion depth 

### ISSUE THREAD

**Issue by shaunharker** [ramonhagenaars/jsons#46]
```python
import jsons
import numpy as np
jsons.dumps(np.array([0]))
```

causes the following error message to repeat:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/anaconda3/lib/python3.7/site-packages/jsons/_main_impl.py", line 63, in dump
    return serializer(obj, cls=cls, **kwargs_)
  File "/usr/local/anaconda3/lib/python3.7/site-packages/jsons/serializers/default_object.py", line 56, in default_object_serializer
    **kwargs_)
  File "/usr/local/anaconda3/lib/python3.7/site-packages/jsons/serializers/default_dict.py", line 25, in default_dict_serializer
    strip_nulls=strip_nulls, **kwargs)
  File "/usr/local/anaconda3/lib/python3.7/site-packages/jsons/_main_impl.py", line 65, in dump
    raise SerializationError(str(err))
jsons.exceptions.SerializationError: maximum recursion depth exceeded in comparison
```

**Comment by ramonhagenaars** [comment]
Hi shaunharker. External libraries such as ``numpy`` are not supported by ``jsons`` yet. I'm thinking of some way to provide extensions in order to support popular libraries like ``numpy``.

For the time being, you could add the following lines to your code to allow (de)serializing ``numpy`` arrays:

```python
import numpy as np
import jsons

jsons.set_serializer(lambda obj, *_, **__: list(obj), np.ndarray)
jsons.set_deserializer(lambda obj, *_, **__: np.array([obj]), np.ndarray)
```

After having added this, you can now dump arrays:

```python
arr = np.array([1, 2, 3])
dumped = jsons.dump(arr)
```

And load them back again:

```python
loaded = jsons.load(dumped, np.ndarray)
```

**Comment by shaunharker** [comment]
Got it. Is it possible to raise a more informative exception (e.g. "X is not jsons-able")?

**Comment by RHagenaars** [comment]
Yes, there are some cases that can definitely use more informative exception messages.

**Comment by ramonhagenaars** [comment]
Yes, there are some cases that can definitely use more informative exception messages.

**Comment by ramonhagenaars** [comment]
Fixes for related issues [Issue#50](https://github.com/ramonhagenaars/jsons/issues/50) and [Issue#51](https://github.com/ramonhagenaars/jsons/issues/51) are deployed in version [0.8.8](https://pypi.org/project/jsons/).
